pkg/loop/internal/datasource.go

```diff
@@ -26,6 +27,10 @@ func (d *dataSourceClient) Observe(ctx context.Context, timestamp types.ReportTi
 	reply, err := d.grpc.Observe(ctx, &pb.ObserveRequest{
 		ReportTimestamp: pbReportTimestamp(timestamp),
 	})
+	// return nil value for NOOP (only affects gas price data source)
+	if errors.Is(err, median.ErrNOOPDataSource) {
```

Comment:
@augustbleeds This still feels wrong: this `dataSource` abstraction is totally generic, and we're introducing some special error handling of the NOOPDataSource type here.

Could the noop data source just return `nil, nil` instead, therefore removing the need for this?

Reply:
sounds good, I incorporated the changes